beacon-chain/types/crystallized_state.go

```diff
@@ -294,11 +289,19 @@ func (c *CrystallizedState) NewStateRecalculations(aState *ActiveState, slotNumb
 		TotalDeposits:              nextCycleBalance,
 	})
 
+	// Entering new dynasty transition.
+	if newCrystallizedState.isDynastyTransition(block.SlotNumber()) {
+		log.Info("Entering dynasty transition")
+		newCrystallizedState, err = c.newDynastyRecalculations(block.ParentHash())
```

Comment:
Since we're doing all the state recalculation in one go, this can just return the new fields for `DynastySeed` and `ShardAndCommitteesForSlots` since we're instantiating a new `CrystallizedState` at the end of this function anyways.